cdap-cli-tests/src/test/java/co/cask/cdap/cli/AuthorizationCLITest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli;
+
+import co.cask.cdap.StandaloneTester;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.twill.LocalLocationFactory;
+import co.cask.cdap.gateway.handlers.InMemoryAuthorizer;
+import co.cask.cdap.internal.test.AppJarHelper;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.proto.security.Role;
+import co.cask.common.cli.CLI;
+import com.google.common.base.Throwables;
+import com.google.common.io.Files;
+import org.apache.twill.filesystem.Location;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tests authorization CLI commands. These tests are in their own class because they need authorization enabled.
+ */
+public class AuthorizationCLITest extends CLITestBase {
+  private static Location extensionJar;
+  static {
+    LocalLocationFactory locationFactory = new LocalLocationFactory(Files.createTempDir());
```

Comment:
Had to do it because I want to use the ``@ClassRule StandaloneTester``, and want to create this extension jar before that. If I use a ``TemporaryFolder`` here, there is no guarantee that it is created before the ``@ClassRule StandaloneTester``. It always fails with "The temporary folder has not yet been created". However, maybe I can make it a non-final ``@ClassRule StandaloneTester`` and instantiate it in ``@BeforeClass``. Not sure how that will work. Will try.